File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start = 0
        size = 0
        maxLen = 99999
        currSum = 0
        
        for i in range(len(nums)):
            currSum += nums[i]
            while currSum >= target:
                maxLen = min(maxLen, i - start + 1);
                currSum -= nums[start];
                start += 1
        if maxLen == 99999:
            maxLen = 0
        return maxLen
                

File: 209-minimum-size-subarray-sum/209-minimum-size-subarray-sum.cpp
class Solution
{
    public:
        int minSubArrayLen(int target, vector<int> &nums)
        {
            int currSum = 0, minLen = 1e9, s = 0;
            for (int e = 0; e < nums.size(); e++)
            {
                currSum += nums[e];
                while (currSum >= target)
                {
                    minLen = min(minLen, e - s + 1);
                    currSum -= nums[s++];
                }
            }
            return minLen == 1e9 ? 0 : minLen;
        }
};